obp/ope/estimators_tuning.py

```diff
@@ -0,0 +1,956 @@
+            raise TypeError("lambdas must be a list")
+
+    def estimate_policy_value(
+        self,
+        reward: np.ndarray,
+        action: np.ndarray,
+        pscore: np.ndarray,
+        action_dist: np.ndarray,
+        position: Optional[np.ndarray] = None,
+        **kwargs,
+    ) -> np.ndarray:
+        """Estimate policy value of an evaluation policy.
+
+        Parameters
+        ----------
+        reward: array-like, shape (n_rounds,)
+            Reward observed in each round of the logged bandit feedback, i.e., :math:`r_t`.
+
+        action: array-like, shape (n_rounds,)
+            Action sampled by a behavior policy in each round of the logged bandit feedback, i.e., :math:`a_t`.
+
+        pscore: array-like, shape (n_rounds,)
+            Action choice probabilities by a behavior policy (propensity scores), i.e., :math:`\\pi_b(a_t|x_t)`.
+
+        action_dist: array-like, shape (n_rounds, n_actions, len_list)
+            Action choice probabilities by the evaluation policy (can be deterministic), i.e., :math:`\\pi_e(a_t|x_t)`.
+
+        position: array-like, shape (n_rounds,), default=None
+            Positions of each round in the given logged bandit feedback.
+
+        Returns
+        ----------
+        V_hat: float
+            Estimated policy value (performance) of a given evaluation policy.
+
+        """
+        if not isinstance(reward, np.ndarray):
+            raise ValueError("reward must be ndarray")
+        if not isinstance(action, np.ndarray):
+            raise ValueError("action must be ndarray")
+        if not isinstance(pscore, np.ndarray):
+            raise ValueError("pscore must be ndarray")
+
+        check_ope_inputs(
+            action_dist=action_dist,
+            position=position,
+            action=action,
+            reward=reward,
+            pscore=pscore,
+        )
+        if position is None:
+            position = np.zeros(action_dist.shape[0], dtype=int)
+
+        # tune the clipping hyperparameter
+        self.estimated_mse_upper_bound_dict = dict()
+        for lambda_ in self.lambdas:
+            estimated_mse_upper_bound = InverseProbabilityWeighting(
+                lambda_=lambda_
+            )._estimate_mse_upper_bound(
+                reward=reward,
+                action=action,
+                position=position,
+                pscore=pscore,
+                action_dist=action_dist,
+            )
+            self.estimated_mse_upper_bound_dict[lambda_] = estimated_mse_upper_bound
+        self.best_lambda_ = min(
+            self.estimated_mse_upper_bound_dict.items(), key=lambda x: x[1]
+        )[0]
+
+        return (
+            InverseProbabilityWeighting(lambda_=self.best_lambda_)
+            ._estimate_round_rewards(
+                reward=reward,
+                action=action,
+                position=position,
+                pscore=pscore,
+                action_dist=action_dist,
+            )
+            .mean()
```

Comment:
(Question)
Is there a reason why you don't use `InverseProbabilityWeighting().estimate_policy_value()`?

Reply:
Fixed. Thanks!